packages/kbn-esql-validation-autocomplete/src/autocomplete/helper.ts

```diff
@@ -80,3 +80,15 @@ export function removeQuoteForSuggestedSources(suggestions: SuggestionRawDefinit
     text: d.text.startsWith('"') && d.text.endsWith('"') ? d.text.slice(1, -1) : d.text,
   }));
 }
+
+export function getSupportedTypesForFunction(
+  fnDef: FunctionDefinition | undefined,
+  previousType: string
+) {
+  // Retrieve list of all 'right' supported types that match the left hand side of the function
+  return fnDef && Array.isArray(fnDef?.signatures)
+    ? fnDef.signatures
+        .filter(({ params }) => params.find((p) => p.name === 'left' && p.type === previousType))
+        .map(({ params }) => params[1].type)
+    : [previousType];
+}
```

Comment:
Does this apply only to binary operators? If so, I wonder if we could name it more specifically

Reply:
Renamed here [`4d1b7a7` (#188600)](https://github.com/elastic/kibana/pull/188600/commits/4d1b7a71b5092a05381063d1609ec6187dedc2a8)